addons/controllers/packageinstallstatus_controller.go

```diff
@@ -0,0 +1,442 @@
+	if clusterRole == clusterRoleWorkload && clusterBootstrap.Spec.Kapp != nil {
+		conditionChanged, err := r.reconcileClusterBootstrapStatus(ctx, r.Client, &clusterBootstrap.Status, cluster.Name, clusterRole, clusterBootstrap.Spec.Kapp.RefName, cluster.Namespace, log)
+		if err != nil {
+			// Note: just log the error and proceed with updating the ClusterBootstrapStatus for other packages in one update operation
+			log.Error(err, fmt.Sprintf("failed to reconcile PackageInstallStatus for package '%s/%s'", cluster.Namespace, clusterBootstrap.Spec.Kapp.RefName))
+		}
+		statusChanged = statusChanged || conditionChanged
+	}
+
+	if !statusChanged {
+		log.Info("no change in the reconciliation status of any of the core/additional packages in the cluster")
+		return nil
+	}
+
+	if err := r.Client.Status().Update(ctx, clusterBootstrap); err != nil {
+		return err
+	}
+
+	log.Info("Successfully reconciled ClusterBootstrapStatus")
+	return nil
+}
+
+// reconcileClusterBootstrapStatus reconciles clusterBootstrapStatus by setting conditions corresponding to all core/additional packages
+// The Status update happens in the caller function
+func (r *PackageInstallStatusReconciler) reconcileClusterBootstrapStatus(
+	ctx context.Context,
+	c client.Client,
+	clusterBootstrapStatus *runtanzuv1alpha3.ClusterBootstrapStatus,
+	clusterName string,
+	clusterRole ClusterRole,
+	pkgName string,
+	pkgiNamespace string,
+	log logr.Logger) (bool, error) {
+
+	var pkgiName, pkgShortname string
+
+	Client := c
+	// controller-runtime's client (r.Client) is not able to find aggregated api server resources.
+	// That's why we need to use r.aggregatedAPIResourcesClient for fetching packageMetadata inside the management cluster.
+	// This include packages belonging to the management cluster and workload clusters' kapp controller package which
+	// is installed in the workload cluster's namespace in the management cluster.
+	// For all other packages belonging to the workload cluster, remoteClient should be used for fetching packageMetadata
+	if clusterRole == clusterRoleManagement || strings.Contains(pkgName, kappCtrlPkgPrefix) {
+		Client = r.aggregatedAPIResourcesClient
+	}
+
+	// Note: pkgi name is determined from pkg.RefName. That's why we rely on the existence of package metadata here
+	pkgRefName, pkgVersion, err := util.GetPackageMetadata(ctx, Client, pkgName, pkgiNamespace)
+	if pkgRefName == "" || pkgVersion == "" || err != nil {
+		log.Error(err, fmt.Sprintf("unable to fetch Package.Spec.RefName or Package.Spec.Version from Package '%s/%s'", pkgiNamespace, pkgName))
+		return false, err
+	}
+	pkgShortname = strings.Split(pkgRefName, ".")[0]
+
+	// package install name for core/additional packages in the workload cluster follows the <cluster name>-<addon short name> naming convention
+	// for management cluster, the naming convention for core/additional package installs is just <addon short name>
```

Comment:
question: I think we use always use `util.GeneratePackageInstallName()` for pkgi names regardless of the cluster role ?

Reply:
Ok, so maybe cause my management cluster I tested with was not utkg, I still see "antrea", "metrics-server", etc. Are we guaranteed to have ```<cluster name>-<addon short name>``` for PackageInstalls in the management cluster now?

Reply:
https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/clusterbootstrap_controller.go#L370

Reply:
I did a search in code(clusterbootstrap_controller.go) for all PKGI related part, I see it always invokes `util. GeneratePackageInstallName()`. That is why I was questioning about `the naming convention for core/additional package installs is just <addon short name>`.

Reply:
Yes, as you mentioned, the clusterbootstrap code is using ```GeneratePackageInstallName()``` irrespective of the cluster type. But at least for current tkgm clusters, packages are made available as part of core repository; someof which still to be made available as part of management repository for utkg and those don't have cluster name prefix. Need to clarify this with @vijaykatam

Reply:
Changed all cluster types to use ```<clustername>-<addonName>``` convention